datalake/src/main/java/com/sequenceiq/datalake/cm/ClouderaManagerRangerUtil.java

```diff
@@ -118,17 +109,61 @@ public boolean isCloudIdMappingSupported(String stackCrn) throws ApiException {
         return isCloudIdMappingSupported(rolesResourceApi, clusterName, rangerUserSyncRoleName);
     }
 
-    public Optional<ApiCommand> setAzureCloudIdentityMapping(String stackCrn, Map<String, String> azureUserMapping) throws ApiException {
+    private Map<String, String> userMappingStrToMap(String userMappingStr) {
+        userMappingStr = userMappingStr.trim();
+        if (userMappingStr.isEmpty()) {
+            return Collections.emptyMap();
+        }
+        Map<String, String> ret = new HashMap<>();
+        String[] mappings = userMappingStr.split(";");
+        Arrays.stream(mappings).forEach(mapping -> {
+            String[] entry = mapping.split("=");
```

Comment:
what happens if there is no `=` sign? I think it would throw an exception when you try to assign value to `val`. Shouldn't we handle this somehow?

Reply:
Right. Now handles the lack of an equals sign by logging a warning and skipping the entry.